zerver/lib/attachments.py

```diff
@@ -160,6 +162,24 @@ def validate_attachment_request(
         # can access it.
         return True, attachment
 
+    message_channel_ids = set()
+    for message in messages:
+        if message.is_stream_message():
+            message_channel_ids.add(message.recipient.type_id)
+
+    # We only fetch channels with public history, since those are the
+    # only channels that user groups can grant content access to.
```

Comment:
Hmm, is that exactly how it works? Or can groups grant content access to previously received messages in a protected history channel?

Reply:
Yes, you are right, fixed!

Reply:
TODO: I need to add a test for this

Reply:
Done